test/perf/perf_test.go

```diff
@@ -95,3 +138,105 @@ func initLogger() (logr.Logger, *os.File, error) {
 	fmt.Printf("configured logger to write messages in '%s'\n", out.Name())
 	return logger, out, nil
 }
+
+// createMultipleSignups creates signups by batch (see `config.GetUserBatchSize()`) and monitors the CPU and memory while the
+// provisioning is in progress. Logs the max CPU and memory during captured during each batch by polling the `/metrics`
+// endpoint in a separate go routine.
+func createSignupsByBatch(t *testing.T, ctx *framework.Context, hostAwait *wait.HostAwaitility, memberAwait *wait.MemberAwaitility, config Configuration, logger logr.Logger) {
+
+	require.Equal(t, 0, config.GetUserCount()%config.GetUserBatchSize(), "number of accounts must be a multiple of %d", config.GetUserBatchSize())
+	hostMetricsRoute, err := hostAwait.SetupRouteForService("host-operator-metrics", "/metrics")
+	require.NoError(t, err)
+
+	t.Logf("provisionning %d accounts by batch of '%d", config.GetUserCount(), config.GetUserBatchSize())
+
+	hostOperatorPod, err := hostAwait.GetHostOperatorPod()
+	require.NoError(t, err)
+	memberOperatorPod, err := memberAwait.GetMemberOperatorPod()
+	require.NoError(t, err)
+
+	for b := 0; b < config.GetUserCount()/config.GetUserBatchSize(); b++ { // provisioning users by batch of `config.GetUserBatchSize()` (eg: 100)
+
+		signups := make([]toolchainv1alpha1.UserSignup, config.GetUserBatchSize())
+		start := time.Now()
+
+		for i := 0; i < config.GetUserBatchSize(); i++ {
+			n := (b*config.GetUserBatchSize() + i)
```

Comment:
it's just a detail, but you don't need the parenthesis here:
```suggestion
			n := b*config.GetUserBatchSize() + i
```